pkg/aws/tools/generator.go

```diff
@@ -0,0 +1,267 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package aws // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/aws/tools"
+
+import (
+	"math"
+	"math/rand/v2"
+	"sort"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/aws/cloudwatch/histograms"
+)
+
+type DistributionConfig struct {
+	Name       string
+	SampleSize int
+	Params     map[string]float64
+}
+
+type GeneratedDataset struct {
+	Name              string
+	Data              []float64
+	ActualPercentiles map[float64]float64
+	Histograms        []histograms.HistogramTestCase
+}
+
+func GenerateTestData() []GeneratedDataset {
+	return GenerateTestDataWithSeed(42)
+}
+
+func GenerateTestDataWithSeed(seed uint64) []GeneratedDataset {
+	configs := []DistributionConfig{
+		{Name: "LogNormal", SampleSize: 2000, Params: map[string]float64{"mu": 2.0, "sigma": 0.5}},
+		{Name: "Weibull", SampleSize: 2000, Params: map[string]float64{"shape": 2.0, "scale": 100.0}},
+		{Name: "Normal", SampleSize: 2000, Params: map[string]float64{"mean": 50.0, "stddev": 15.0}},
+		{Name: "Gamma", SampleSize: 2000, Params: map[string]float64{"shape": 2.0, "rate": 0.1}},
+	}
+
+	var datasets []GeneratedDataset
+	for _, config := range configs {
+		// each dataset gets a copy of the same rand so that they don't interfere with each other
+		rng := rand.New(rand.NewPCG(seed, seed))
```

Comment:
How much randomness do we need and how significant is it? what happens if each config shares the same rand?